dart/lib/src/sentry_options.dart

```diff
@@ -76,7 +76,19 @@ class SentryOptions {
 
   /// Turns debug mode on or off. If debug is enabled SDK will attempt to print out useful debugging
   /// information if something goes wrong. Default is disabled.
-  bool debug = false;
+  bool get debug => _debug;
+
+  set debug(bool newValue) {
+    _debug = newValue;
+    if (_debug == true && logger == noOpLogger) {
+      _logger = dartLogger;
+    }
+    if (_debug == false && logger == dartLogger) {
```

Comment:
```suggestion
    if (_debug == false && logger != noOpLogger) {
```

maybe its better?

Reply:
That would override the logger which the user has set. We only want to override the logger in case it wasn't altered by the user.

Or am I missing something?

Reply:
yeah makes sense, it does not make sense to reset to default value